cve_bin_tool/parsers/javascript.py

```diff
@@ -23,6 +25,17 @@ def run_checker(self, filename):
                 vendor = None
             if vendor is not None:
                 yield from vendor
+
+            # If there is no "dependencies" array then this file is likely not in a format we
+            # can parse.  Log warning and abort.
+            # npm generates a similarly named .package-lock.js file (note the starting `.`)
+            # that will trigger this
+            if "dependencies" not in data.keys():
+                self.logger.warning(
+                    f"Cannot parse {self.filename}: no dependencies array found."
+                )
```

Comment:
That looks good. The only thing I can think of is: is there a case where you can have more than 1 "dependencies" keys? If so, would it be better to also add the product and version to the warning message? (Feel free to ignore if too dumb of a comment)

Reply:
Hm, in theory people could give us just about anything so it would be possible, but I haven't seen any tool generate this yet or anyone complain about us not parsing it.  I'll put a note about it when I file an issue about handling the npm package locks, though, since it might be relevant then.